scripts/globals/chocobo_digging.lua

```diff
@@ -0,0 +1,255 @@
+
+            if (Chance < (RItemAbundance + DIG_ABUNDANCE_BONUS)) then
+            
+                local RItemID = RandomItem[1];
+                local RItemReq = RandomItem[3];
+                
+                -- rank 1 is burrow, rank 2 is bore (see DIGABILITY at the top of the file)
+                local DigAbility = player:getSkillRank(SKILL_DIG);
+                
+                local Mod = player:getMod(MOD_EGGHELM);
+                
+                if (RItemReq == DIGREQ_NONE) then
+                    ItemID = RItemID;
+                elseif (RItemReq == DIGREQ_BURROW and DigAbility == DIGABILITY_BURROW) then
+                    ItemID = RItemID;
+                elseif (RItemReq == DIGREQ_BORE and DigAbility == DIGABILITY_BORE) then
+                    ItemID = RItemID;
+                elseif (RItemReq == DIGREQ_MODIFIER and Mod) then
+                    ItemID = RItemID;
+                elseif (RItemReq == DIGREQ_NIGHT and VanadielTOTD() == TIME_NIGHT) then
+                    ItemID = RItemID;
+                end
+                
+                -- Let's see if the item should be obtained in this zone with this weather
+
+                if ((weather == 4 or weather == 5) and ItemID == 4096) then
+                  ItemID = ItemID;
+                elseif ((weather == 6 or weather == 7) and ItemID == 4097) then
+                  ItemID = ItemID;
+                elseif ((weather == 8 or weather == 9) and ItemID == 4098) then
+                  ItemID = ItemID;
+                elseif ((weather == 10 or weather == 11) and ItemID == 4099) then
+                  ItemID = ItemID;
+                elseif ((weather == 12 or weather == 13) and ItemID == 4100) then
+                  ItemID = ItemID;
+                elseif ((weather == 14 or weather == 15) and ItemID == 4101) then
+                  ItemID = ItemID;
+                elseif ((weather == 16 or weather == 17) and ItemID == 4102) then
+                  ItemID = ItemID;
+                elseif ((weather == 18 or weather == 19) and ItemID == 4103) then
+                  ItemID = ItemID;
+                else
+                  ItemID = 0;
+                end
+                
+                -- If the item is a crystal, we need to check for a double weather effect (Which results in a crystal cluster, not a single crystal)
+                if (ItemID >= 4096 or ItemID <= 4103) then
+                  if (zoneWeather == "WEATHER_DOUBLE") then
+                    ItemID = ItemID + 8;
+                  elseif (zoneWeather == "WEATHER_SINGLE") then
+                    ItemID = ItemID;
+                  else
+                    ItemID = 0;
+                  end
+                end
+                
+                -- If the item is an elemental ore, we need to check if the requirements are met
+                
+                if ((ItemID >= 1255 and ItemID <= 1262) and weather > 1 and (moon >= 10 and moon <= 40) and SkillRank >= 7) then
```

Comment:
And this was done by me, for reasons above. This is easier to read for people who want to work on it (Who may not necessarily be as experienced of programmers as others), than reading id { id, id, id } , id { id, id, id } }